mp4parse/src/lib.rs

```diff
@@ -1682,6 +1711,62 @@ fn read_stsd<T: Read>(src: &mut BMFFBox<T>, track: &mut Track) -> Result<SampleD
     })
 }
 
+fn read_sinf<T: Read>(src: &mut BMFFBox<T>) -> Result<ProtectionSchemeInfoBox> {
+    let mut sinf = ProtectionSchemeInfoBox::default();
+
+    let mut iter = src.box_iter();
+    while let Some(mut b) = iter.next_box()? {
+        match b.head.name {
+            BoxType::OriginalFormatBox => {
+                let frma = read_frma(&mut b)?;
+                sinf.code_name = frma;
+            },
+            BoxType::SchemeInformationBox => {
+                // We only need tenc box in schi box so far.
+                let schi_tenc = read_schi(&mut b)?;
+                sinf.tenc = schi_tenc;
+            },
```

Comment:
Do we depend on the `tenc` data? If so we should probably also reject streams with a `sinf` without an `schi`.